components/org.wso2.carbon.crypto.provider/src/test/java/org/wso2/carbon/crypto/provider/SymmetricKeyInternalCryptoProviderTest.java

```diff
@@ -0,0 +1,225 @@
+    private static final String CIPHER_TEXT_GCM = "eyJjIjoiZXlKamFYQm9aWElpT2lJMFRtSnpPWEJIUkdOeFEzaG1PVUZEYkRWalpscDRRMFpPVkVoa2NHbFRibFF4TkQwaUxDSnBibWwwYVdGc2FYcGhkR2x2YmxabFkzUnZjaUk2SWtweWNYTlFWMDl5UldVclFURkhUMEpPTWs1YVZYZEJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVQwaUxDSnJaWGxKWkNJNklsb3pNaTkxTjNwVmMzSktaMVo1Tnk5cE9YWmpXRXRvV1VaaVpYZHRTMnRKTTFsRk56UjVkMngyYm5NOUluMD0iLCJ0IjoiQUVTL0dDTS9Ob1BhZGRpbmciLCJpdiI6IkpycXNQV09yRWUrQTFHT0JOMk5aVXdBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQT0iLCJraWQiOiJaMzIvdTd6VXNySmdWeTcvaTl2Y1hLaFlGYmV3bUtrSTNZRTc0eXdsdm5zPSJ9";
+    private static final String CIPHER_TEXT_GCM_FOR_OLD_SECRET = "eyJjIjoiZXlKamFYQm9aWElpT2lKelowVlVSVEZpY1RaMFkxWXJiWFl5UTA1eWRGbHdkWEJoVUVsSFdsTnNWVVpFTUQwaUxDSnBibWwwYVdGc2FYcGhkR2x2YmxabFkzUnZjaUk2SWt0NVNXUk5SMDkxUldVcldUVTFkV1FyVEdKNGNGRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVVGQlFVRkJRVUZCUVQwaUxDSnJaWGxKWkNJNklqRTJUUzlzUW5oVmR6ZHBSakJ0V1hodlZHc3hOU3R0VUdOMFpWTm1WRlZOSzNadlpXWldhSGx1Y0c4OUluMD0iLCJ0IjoiQUVTL0dDTS9Ob1BhZGRpbmciLCJpdiI6Ikt5SWRNR091RWUrWTU1dWQrTGJ4cFFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQT0iLCJraWQiOiIxNk0vbEJ4VXc3aUYwbVl4b1RrMTUrbVBjdGVTZlRVTSt2b2VmVmh5bnBvPSJ9";
+
+    @BeforeClass
+    public void init() {
+
+        cryptoProvider = new SymmetricKeyInternalCryptoProvider(SECRET);
+        cryptoProviderWithBackupSecret = new SymmetricKeyInternalCryptoProvider(SECRET, OLD_SECRET, true);
+    }
+
+    // Default tests.
+
+    @Test(description = "Test encryption using the default secret.")
+    public void testEncrypt() throws CryptoException {
+
+        byte[] plainTextBytes = PLAIN_TEXT.getBytes();
+        byte[] encryptedText = cryptoProvider.encrypt(plainTextBytes, AES_ALGORITHM, JCE_PROVIDER);
+        assertEquals(new String(Base64.getEncoder().encode(encryptedText)), CIPHER_TEXT);
+    }
+
+    @Test(description = "Test encryption using a old secret.")
+    public void testEncryptUsingOldSecret() throws CryptoException {
+
+        byte[] plainTextBytes = PLAIN_TEXT.getBytes();
+        byte[] encryptedText = cryptoProvider.encrypt(
+                plainTextBytes, AES_ALGORITHM, JCE_PROVIDER, false, OLD_SECRET);
+        // TODO: Should return CIPHER_TEXT_FOR_OLD_SECRET.
```

Comment:
Either fix this todo or create a git issue for this.

Reply:
Fixed the todo